### PR TITLE
AK: Make Optional<T> explicitly constructible from Optional<U>

### PR DESCRIPTION
**Pull request by IdanHo**

### CODE REVIEW

File: AK/Optional.h [hunk 1]
```diff
@@ -63,6 +66,24 @@ class [[nodiscard]] Optional {
         }
     }
 
+    template<typename U>
+    requires(IsConstructible<T, U const&> && !IsSpecializationOf<T, Optional> && !IsSpecializationOf<U, Optional>) ALWAYS_INLINE explicit Optional(Optional<U> const& other)
+        : m_has_value(other.m_has_value)
+    {
+        if (other.has_value()) {
+            new (&m_storage) T(other.value());
+        }
```
Comment (linusg) [inline AK/Optional.h:75]: curly braces literally *useless* here

File: AK/Optional.h [hunk 2]
```diff
@@ -63,6 +66,24 @@ class [[nodiscard]] Optional {
...
+    }
+
+    template<typename U>
+    requires(IsConstructible<T, U&&> && !IsSpecializationOf<T, Optional> && !IsSpecializationOf<U, Optional>) ALWAYS_INLINE explicit Optional(Optional<U>&& other)
+        : m_has_value(other.m_has_value)
+    {
+        if (other.has_value()) {
+            new (&m_storage) T(other.release_value());
+        }
```
Comment (linusg) [inline AK/Optional.h:84]: Same

File: Userland/Libraries/LibJS/Runtime/Temporal/PlainDateTimePrototype.cpp [hunk 3]
```diff
@@ -537,14 +537,14 @@ JS_DEFINE_NATIVE_FUNCTION(PlainDateTimePrototype::until)
     // 9. Perform ? ValidateTemporalUnitRange(largestUnit, smallestUnit).
     TRY(validate_temporal_unit_range(global_object, *largest_unit, *smallest_unit));
 
-    // 10. Let roundingMode be ? ToTemporalRoundingMode(options, "trunc").
+    // 10. Let roundingMode be ? ToTemporalRoundingMode(options, "trunc").<
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Temporal/PlainDateTimePrototype.cpp:540]: that doesn't seem to be on purpose

File: AK/Optional.h (hunk 1 above)
Comment (IdanHo) [reply AK/Optional.h:75]: They are indeed, I just copied it from the constructor above, should I fix it up as well? :^)

File: Userland/Libraries/LibJS/Runtime/Temporal/PlainDateTimePrototype.cpp (hunk 3 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/Temporal/PlainDateTimePrototype.cpp:540]: Whoops! fixed.

File: AK/Optional.h (hunk 1 above)
Comment (linusg) [reply AK/Optional.h:75]: might as well, yeah :)

**Review by alimpfard — APPROVED** [review]
LGTM.

**Review by linusg — APPROVED** [review]

**Pull request merged**